craft_providers/lxd/lxc.py

```diff
@@ -573,18 +592,65 @@ def launch(
             for config_key in [f"{k}={v}" for k, v in config_keys.items()]:
                 command.extend(["--config", config_key])
 
-        try:
-            self._run_lxc(
-                command,
-                capture_output=True,
-                stdin=StdinType.INTERACTIVE,
-                project=project,
+        # The total times of retrying to launch the same instance per craft-providers.
+        # If parallel lxc failed, the bad instance will be deleted by the lock holder
+        # or any other craft-providers, and the lock will be released.
+        # However, the new instance lock could be held by any craft-providers.
+        # This is used to avoid lock holder dead and all others are blocked.
+        while retry_count < 3:
```

Comment:
Are 3 retries enough for typical cases, considering slow machines?

Reply:
It should not have any retry there if lxc is working. But retry when the instance creation failed. So I think 3 times is enough for network error or bad system configuration.

Reply:
I'm not too worried about having 3 retries here. The types of errors we want to catch should be ephemeral.